src/tokens/Datatoken.ts

```diff
@@ -143,6 +143,36 @@ export class Datatoken {
     return trxReceipt
   }
 
+  /**
+   * ApproveWei
+   * @param {String} dtAddress Datatoken address
+   * @param {String} spender Spender address
```

Comment:
What is this method for? We have a generic `approve` method over in `TokenUtils.ts` and it uses wei

Reply:
providerFeeAmount is always expressed in Wei, so when paying that fee, we need to approve first.
Converting from Wei to Decimal and then back to Wei will most likely cause some rounding errors, which will result in a fail validation in https://github.com/oceanprotocol/contracts/blob/v4main/contracts/templates/ERC20Template.sol#L453 , because the signature is created using the Wei amount

Reply:
as usage, it's here:  https://github.com/oceanprotocol/ocean.js/blob/feature/ddo_schema_4.1.0/test/integration/ComputeFlow.test.ts#L312

(handleOrder function is going to be most likely duplicated in the market, it's not ocean.js specific, but more app)

Reply:
about gasEstimate:     Would be nice, but technically speaking,  
     estimateGas (approve( 0x11, 100))  or 
     estimateGas (approve( 0x11, 1000000000000000))
 costs the same, so you don't need Wei and non-Wei, the result will be the same..

Reply:
This function should be in token utils and indeed the one that already exists has bad comments , it doesn't accept wei, but a number (as string) . It doesn't hurt to have one that takes wei directly just in case.